src/rin2016/module8/rubber_array/Array.java

```diff
@@ -0,0 +1,241 @@
+package rin2016.module8.rubber_array;
+
+public class Array {
+	private int[] array;
+	private int size;
+	
+	public final static int SEARCH_ERROR = -1;
+	private final int DEFAULT_SIZE = 5;
+	
+	public Array() {
+		create(DEFAULT_SIZE);
+		
+	}
+	
+	public Array(final int size) {
+		create(size);
+	}
+	
+	private void create(final int size) {
+		if (size >= 0) {
+			this.array = new int[size];
+			this.size = 0;
+		} else {
+			this.array = new int[DEFAULT_SIZE];
+			this.size = 0;
+		}
+	}
+	
+	public int getElement(final int index) {
+		if(index <= size && index >= 0) {
+			return array[index];
+		}
+		return Integer.MIN_VALUE;
+	}
+	
+	private int[] getArray() {
+		int[] result = new int [size];
+		copy(result, array, size);
+		return result;
+	}
+	
+	public int size() {
+		return this.size;
+	}
+	
+	/*
+	 * Добавляем элемент в конец массива
+	 */
+	public void add(final int element) {
+		if (size == 0) {
+			this.array[size++] = element;
+		} else {
+			add(element, size);
+		}
+	}
+	
+	/*
+	 * Добавляем элемент в массив (по индексу)
+	 */
+	public void add(final int element, final int index) {
+		if (index > 0 && index <= size) {
+			int sizeArray = array.length-1;
+			if (index >= sizeArray) {
+				int newSize = array.length * 2;
+				newSizeArray(newSize, index);
+			}
+			this.array[index] = element;
+			this.size++;
+		}
+	}
+	
+	
+	
+	private int[] copy(final int[] firstArray, 

```

Comment:
Please, rename your arguments to reflect their role in this method. Use such names as "from", "to" or "source", "destination"